src/rpc/methods/chain.rs

```diff
@@ -978,28 +993,155 @@ impl RpcMethod<1> for ChainGetTipSet {
 
     async fn handle(
         ctx: Ctx<impl Blockstore>,
-        (ApiTipsetKey(tipset_key),): Self::Params,
+        (ApiTipsetKey(tsk),): Self::Params,
     ) -> Result<Self::Ok, ServerError> {
-        let ts = ctx
-            .chain_store()
-            .load_required_tipset_or_heaviest(&tipset_key)?;
-        Ok(ts)
+        if let Some(tsk) = &tsk {
+            let ts = ctx.chain_index().load_required_tipset(tsk)?;
+            Ok(ts)
+        } else {
+            // Error message here matches lotus
```

Comment:
Do we need to match the Lotus error? This would actually be a very confusing message for me.

Reply:
Updated, but still keep Lotus error message in parentheses to make RPC parity test easier.